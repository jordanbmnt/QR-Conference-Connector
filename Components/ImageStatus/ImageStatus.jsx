import { Image, StyleSheet, View } from "react-native";

export default ImageStatus = ({ theme }) => {
  return (
    <View style={styles.noContactsBackground}>
      {theme === "noContacts" ? (
        <Image
          style={styles.img}
          source={require("../assets/images/noContacts.png")}
          alt='noContact'
        />
      ) : theme === "invalidQR" ? (
        <Image
          source={require("../assets/images/invalidQR.png")}
          alt='invalidQR'
        />
      ) : theme === "permission" ? (
        <Image
          source={require("../assets/images/permission.png")}
          alt='permission'
        />
      ) : null}
    </View>
  );
};

const styles = StyleSheet.create({
  noContactsBackground: {
    justifyContent: "center",
    alignContent: "center",
    opacity: 0.5,
    height: "80%",
  },
  img: {
    width: 400,
    height: 400,
  },
});
