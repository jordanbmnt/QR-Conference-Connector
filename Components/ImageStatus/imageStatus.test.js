import renderer from "react-test-renderer";
import ImageStatus from "./ImageStatus";

describe("ImageStatus", function () {
  const image = renderer.create(<ImageStatus theme='noContact' />).toJSON();
  it("should render the image component correctly", () => {
    expect(image).toMatchSnapshot();
  });
  it("should be transparent", function () {
    expect(image.props.style.opacity).toBe(0.5);
  });
  it("should render the imageStatus component correctly", () => {
    expect(image).toMatchSnapshot();
  });
});
