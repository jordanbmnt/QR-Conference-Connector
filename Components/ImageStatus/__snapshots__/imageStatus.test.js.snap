// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageStatus should render the image component correctly 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "height": "80%",
      "justifyContent": "center",
      "opacity": 0.5,
    }
  }
/>
`;

exports[`ImageStatus should render the imageStatus component correctly 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "height": "80%",
      "justifyContent": "center",
      "opacity": 0.5,
    }
  }
/>
`;
